.github/workflows/docs.yml

```diff
@@ -0,0 +1,191 @@
+name: Docs
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - "docs/**"
+      - "solidity/scripts/generate-api-docs.js"
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  docs-detect-changes:
+    runs-on: ubuntu-latest
+    outputs:
+      path-filter: ${{ steps.filter.outputs.path-filter }}
+    steps:
+      - uses: actions/checkout@v2
+        if: github.event_name == 'pull_request' 
+      - uses: dorny/paths-filter@v2
+        if: github.event_name == 'pull_request' 
+        id: filter
+        with:
+          filters: | # TODO: Remove '|(.github/workflows/docs.yml)' once workflow gets tested
+            path-filter:
+              - './((docs/**)|(solidity/scripts/generate-api-docs.js)|(.github/workflows/docs.yml))'
+
+  generate-pdf-from-tex:
+    runs-on: ubuntu-latest
+    needs: docs-detect-changes
+    if: |
+      github.event_name != 'pull_request'
+        || needs.docs-detect-changes.outputs.path-filter == 'true'
+    env:
+      DOCS_DIR: ${{ github.workspace }}/tmp/docs
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Create temporary docs directory
+        run: mkdir -p $DOCS_DIR
+
+      # - name: Generate TeX document files
+      #   uses: addnab/docker-run-action@v3 # Runs shell commands within specified container
+      #   with:
+      #     registry: ghcr.io
+      #     image: keepnetwork/texlive:3
+      #     options: -v ${{ github.workspace }}:/volume-dir
+      #     run: |
+      #       mkdir /volume-dir/tmp/docs/img
+      #       cd /volume-dir/docs
+      #       make clean
+      #       make docs
+
+      - name: Compile TeX files to PDFs
```

Comment:
Thoughts on running this job in the following way:
```
    runs-on: ubuntu-latest
    container: keepnetwork/texlive:3
    defaults:
      run:
        working-directory: ./docs
    steps:
      - uses: actions/checkout@v2
      - run: make clean 
      - run: make docs
      # now publish PDF to bucket
```

Reply:
I did end up applying slightly modified solution. I had to add moving generated file to separate folder, as later action publishing the files to bucket expects the files to be in `build-folder` and providing the path to `docs` there would upload too much.